'All data' export option: Remove records that occur twice.

Sanny26 opened the pull request:
### Proposed changes in this pull request


This commit removes extra duplicate files that are downloaded when 
'all data' is downloaded.
Fix bug https://github.com/Cadasta/cadasta-platform/issues/923

### When should this PR be merged

Anytime

### Risks
None

### Follow up actions
None


### Checklist (for reviewing)

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature** has the manual testing spreadsheet been updated with instructions for manual testing?


#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

oliverroick commented on cadasta/organization/forms.py:
```diff
@@ -489,15 +489,15 @@ def get_file(self):
             with ZipFile(path, 'a') as myzip:
-                myzip.write(data_path, arcname='data.xlsx')
-                myzip.write(shp_path, arcname='data-shp.zip')
+                with ZipFile(res_path, 'r') as res_zip:
```
Here you're writing all the resources into the shp file zip. I would do it the other way around because it's likely that you and up with many more resources files that shp files.

Sanny26 replied:
I assumed that only one resource file will be there for every download. So, I moved this file to zip. I'l update the PR to copy the shp files to resource files.

Sanny26 commented:
@oliverroick Sorry for not asking if the problem can be solved. I will take care next time. Should I close the PR?
However, I have doubt regarding the files being copied.
It is mentioned in the issue that data-shp.zip has location.csv, parties.csv and relationships.csv.  Data.xlsx also contains the same data. So, I removed the .xlsx file as this data already exists in data-shp.zip.
Did I misunderstand the issue?

oliverroick commented:
The problem is that `locations.csv` does not contain geometries itself, the geometry is only in SHP. So in order to get the geometries for the locations, you need to read the SHP first and then join the location data somehow. 

Like I said, we need to discuss internally what the best way to approach this is. You can keep the PR open for the time being, we will get back to you.

seav commented:
Instead of not including the XLSX file, I think it would be better to not include the CSV files. This way, there is no duplication of data, but the SHP geometries are still found also in the XLSX file.

Sanny26 commented:
@seav Can I continue to fix this bug following the approach that you suggested in your comment?

Sanny26 commented:
@oliverroick I updated the PR with these changes
- Moved the shp files into resources.zip
- Downloading the data.xlsx file and removing .csv files from the shp zip file.

oliverroick commented on cadasta/organization/forms.py:
```diff
@@ -494,10 +494,14 @@ def get_file(self):
             data_path, _ = xls_exporter.make_download(file_name + '-xls')
             shp_path, _ = shp_exporter.make_download(file_name + '-shp')
-
+            duplicates = ['locations.csv', 'parties.csv', 'relationships.csv.']
```
`relationships.csv` is still included in the export because the name has a trailing `.` here

Sanny26 commented:
updated as suggested

seav commented:
Hi @Sanny26, this PR should be good to go. You just need to rebase onto the latest master branch.
